### PR TITLE
Moved over deletion methods from the mobile version of the library

### PR DESCRIPTION
**Pull request by pelle**
Note I am uncertain how to run tests, so please run tests before merging in.

### CODE REVIEW

**Comment by wjmelements** [comment]
Can you link the mobile version referenced in the title?

File: UPTEthereumSignerTests/UPTHDSignerSpec.m [hunk 1]
```diff
@@ -75,6 +75,24 @@
...
+    describe(@"Deletion", ^{
+        it(@"Can delete", ^{
+            [UPTHDSigner createHDSeed:UPTHDSignerProtectionLevelPromptSecureEnclave
+                   rootDerivationPath:UPORT_ROOT_DERIVATION_PATH
+                             callback:^(NSString *ethAddress, NSString *publicKey, NSError *error) {
+                                 NSLog(@"eth address: %@ . for public key: %@", ethAddress, publicKey);
+                                 expect(error).to.beNil();
+                                 [UPTHDSigner deleteSeed:ethAddress callback:^(BOOL deleted, NSError *error) {
+                                     XCTAssertTrue(deleted);
+                                     XCTAssertNil(error);
```
Comment (wjmelements) [inline UPTEthereumSignerTests/UPTHDSignerSpec.m:89]: This test should also verify that the seed was deleted.

**Review by wjmelements — CHANGES_REQUESTED** [review]

**Comment by wjmelements** [comment]
@pelle To run the tests, open the workspace in XCode and run Test on the UPTEthereumSigner target.

I have ran the tests on this branch; they pass.

**Comment by wjmelements** [comment]
Here is a doc from Apple on how to run tests in XCode https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/testing_with_xcode/chapters/05-running_tests.html

**Comment by pelle** [comment]
@wjmelements This is the local changes made in the app version of this code https://github.com/uport-project/uport-mobile/tree/develop/ios/uPortMobile

The test option is greyed out in my xcode. It may be because I'm on XCode 10.1.

**Review by meaning-matters — APPROVED** [review]
All tests run (I do see failed assertions (as I did on develop branch a few days ago), so assume that's normal; will keep that in mind check in coming weeks).

**Comment by pelle** [comment]
They all pass for me now @meaning-matters

**Pull request merged**